src/components/CollaboratorModal/components/RemoveCollaboratorSubmodal.tsx

```diff
@@ -0,0 +1,91 @@
+import {
+  ModalHeader,
+  ModalBody,
+  Text,
+  Stack,
+  ModalProps,
+  ModalOverlay,
+  Modal,
+  ModalContent,
+} from "@chakra-ui/react"
+import { Button, ModalCloseButton } from "@opengovsg/design-system-react"
+import { useParams } from "react-router-dom"
+
+import { useLoginContext } from "contexts/LoginContext"
+
+import { useDeleteCollaboratorHook } from "hooks/collaboratorHooks"
+
+import { Collaborator } from "types/collaborators"
+
+import { TEXT_FONT_SIZE } from "../constants"
+
+interface RemoveCollaboratorSubmodalProps extends Omit<ModalProps, "children"> {
+  userToDelete: Collaborator
+  onDeleteComplete: () => void
+}
+
+export const RemoveCollaboratorSubmodal = ({
+  userToDelete,
+  onDeleteComplete,
+  ...props
+}: RemoveCollaboratorSubmodalProps): JSX.Element => {
+  const { email } = useLoginContext()
+  const { siteName } = useParams<{ siteName: string }>()
+  const isUserDeletingThemselves = email === userToDelete?.email
+  const { onClose } = props
+
+  const {
+    mutateAsync: deleteCollaborator,
+    isLoading: isDeleteCollaboratorLoading,
+  } = useDeleteCollaboratorHook(siteName)
+
+  return (
+    <Modal {...props}>
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader>Remove collaborator?</ModalHeader>
+        <ModalCloseButton />
+        <ModalBody mb="44px">
+          {isUserDeletingThemselves ? (
+            <Text>
+              Once you remove yourself as a collaborator, you will no longer be
+              able to make any changes to this site.
+            </Text>
+          ) : (
+            <Text fontSize={TEXT_FONT_SIZE}>
+              <Text as="span">Once you remove</Text>
+              <Text color="danger.700" as="strong">
+                {" "}
+                {userToDelete?.email}{" "}
+              </Text>
+              <Text as="span">
+                from this site, they will no longer be able to make any changes.
+              </Text>
+            </Text>
+          )}
+
+          <Stack spacing={4} direction="row" justify="right" mt="36px">
+            <Button variant="clear" color="secondary" onClick={onClose}>
```

Comment:
I think this `onClose` is insufficient, as it can cause the user to be stuck. Steps to reproduce:
1. Ensure you are the only collaborator left on the site
2. Attempt to remove yourself as the collaborator
3. Error message correctly shows that the site needs at least one admin
4. However, the modal still stays on the RemoveCollaboratorSubmodal. Cancelling/Closing the modal and reopening the collaborators modal again still brings the user to the RemoveCollaboratorSubmodal.

In addition, if the user accidentally clicks on the remove collaborator trash bin icon, there is no way for the user to cancel to go back to the original modal. The user is forced to remove the collaborator.

We should bring the user back to the MainSubmodal on close so that the user is not stuck.

Reply:
yea, thanks for the catch. 2 fold fix, disable if only 1 admin left + reset state on close completion to show main modal